### PR TITLE
release/1.5.0: site config and doc updates round 2 (S4, aws-pcluster, Discover, Nautilus, JCSDA CI containers)

### PR DESCRIPTION
**Pull request by climbfuji**
### Summary

Round 2 of the site config and documentation updates for release 1.5.0: S4, AWS Parallelcluster, Discover, Nautilus, JCSDA CI containers.

### Testing

spack-stack-1.5.0 installed on all platforms and basic tests run as described in https://github.com/JCSDA/spack-stack/issues/765

### Applications affected

All applications using spack-stack-1.5.0 on these platforms

### Systems affected

See above

### Dependencies

n/a

### Issue(s) addressed

Working towards https://github.com/JCSDA/spack-stack/issues/765

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

File: configs/sites/discover/packages.yaml [hunk 1]
```diff
@@ -3,7 +3,6 @@ packages:
     compiler:: [intel@2022.0.1, gcc@10.1.0]
     providers:
       mpi:: [intel-oneapi-mpi@2021.5.0, openmpi@4.1.3]
-    target: [ivybridge]
```
Comment (climbfuji) [inline configs/sites/discover/packages.yaml:6]: No longer needed, all nodes on Discover are now skylake/cascadelake

File: configs/sites/discover/modules.yaml [hunk 2]
```diff
@@ -5,3 +5,5 @@ modules:
     lmod:
       exclude:
       - ecflow
+      include:
+      - python
```
Comment (climbfuji) [inline configs/sites/discover/modules.yaml]: Note to self - fix missing newline

File: configs/sites/s4/modules.yaml [hunk 3]
```diff
@@ -7,3 +7,4 @@ modules:
       - ecflow
       include:
       - mpich
+      - python
```
Comment (climbfuji) [inline configs/sites/s4/modules.yaml]: Fix missing newline

File: configs/containers/docker-ubuntu-clang-mpich.yaml [hunk 4]
```diff
@@ -235,8 +235,6 @@ spack:
         echo "export CXX=clang++" >> /root/.bashrc && \
         echo "export FC=gfortran" >> /root/.bashrc && \
         printf "[credential]\n    helper = cache --timeout=7200\n" >> /root/.gitconfig
-        # Source spack environment script for root
```
Comment (climbfuji) [inline configs/containers/docker-ubuntu-clang-mpich.yaml:238]: Due to an upstream change in spack, these scripts are no longer needed/no longer exist.

File: configs/sites/aws-pcluster/modules.yaml [hunk 5]
```diff
@@ -4,7 +4,6 @@ modules:
     - lmod
     lmod:
       include:
-      # List of packages for which we need modules that are blacklisted by default
```
Comment (climbfuji) [inline configs/sites/aws-pcluster/modules.yaml:7]: DEI DEI

**Comment by AlexanderRichert-NOAA** [comment]
@climbfuji could I trouble you to whitelist the python module for acorn while we're at it? :)
```
    lmod:
      include:
      - python
```
in configs/sites/acorn/modules.yaml. Already tested in the installation.

**Comment by climbfuji** [comment]
> @climbfuji could I trouble you to whitelist the python module for acorn while we're at it? :)
> 
> ```
>     lmod:
>       include:
>       - python
> ```
> 
> in configs/sites/acorn/modules.yaml. Already tested in the installation.

Of course, I will *include* it in the module file.

**Comment by climbfuji** [comment]
> > @climbfuji could I trouble you to whitelist the python module for acorn while we're at it? :)
> > ```
> >     lmod:
> >       include:
> >       - python
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > in configs/sites/acorn/modules.yaml. Already tested in the installation.
> 
> Of course, I will _include_ it in the module file.

Done!

**Review by AlexanderRichert-NOAA — APPROVED** [review]

**Pull request merged**